packages/headless/src/api/search/search/raw.ts

```diff
@@ -1,27 +1,21 @@
-type BaseRaw = {
+export interface Raw {
+  /**
+   * Custom keys that depend on the documents in the index.
+   */
+  [key: string]: unknown;
+
+  /**
+   * @internal @preapproved
```

Comment:
This combination of decorators is to prevent the annotated interface property from being extracted and documented by the tooling.

The search api documentation does not name any properties inside the `raw` object. This makes headless docs consistent with the search api while still providing autocomplete.

Personally, I do not think it is useful to document that names of all properties that _may_ exist.